samples/CppArExample/CppArExample.cpp

```diff
@@ -0,0 +1,353 @@
+#include "GeometryEngine.h"
+#include "PointCloudLayer.h"
+#include "PortalItem.h"
+#include "IntegratedMeshLayer.h"
+#include "LocationDataSource.h"
+
+#include <QUrl>
+#include <QFileInfo>
+#include <QStandardPaths>
+
+#include <QGeoPositionInfoSource>
+
+#include "SimpleMarkerSceneSymbol.h"
+#include "SimpleLineSymbol.h"
+
+using namespace Esri::ArcGISRuntime;
+using namespace Esri::ArcGISRuntime::Toolkit;
+
+CppArExample::CppArExample(QObject* parent):
+  QObject(parent)
+{
+}
+
+CppArExample::~CppArExample()
+{
+}
+
+ArcGISArView* CppArExample::arcGISArView() const
+{
+  return m_arcGISArView;
+}
+
+void CppArExample::setArcGISArView(ArcGISArView* arcGISArView)
+{
+  if (!arcGISArView || arcGISArView == m_arcGISArView)
+    return;
+
+  m_arcGISArView = arcGISArView;
+
+  emit arcGISArViewChanged();
+}
+
+SceneQuickView* CppArExample::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void CppArExample::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+    return;
+
+  m_sceneView = sceneView;
+  emit sceneViewChanged();
+}
+
+// properties for debug mode
+void CppArExample::showPointCloud(bool visible)
+{
+  if (visible)
+    m_arcGISArView->setPointCloudColor(QColor(50, 50, 255));
+  else
+    m_arcGISArView->setPointCloudColor(QColor());
+}
+
+void CppArExample::showPlanes(bool visible)
+{
+  if (visible)
+    m_arcGISArView->setPlaneColor(QColor(255, 0, 0, 10 ));
+  else
+    m_arcGISArView->setPlaneColor(QColor());
+}
+
+// Creates an empty scene with an elevation source.
+// Mode: Full-Scale AR
+void CppArExample::createEmptyScene()
+{
+  // create an empty scene with elevation
+  m_scene = new Scene(this);
+  createSurfaceWithElevation();
+
+  // Set the location data source so we use our GPS location as the originCamera.
+  m_arcGISArView->setOriginCamera(Camera());
+  m_arcGISArView->setTranslationFactor(1.0);
+
+  changeScene(true);
+}
+
+// Creates a scene based on the Streets base map.
+// Mode: Full-Scale AR
+void CppArExample::createStreetsScene()
+{
+  // create a scene with streets
+  m_scene = new Scene(BasemapType::Streets, this);
+  createSurfaceWithElevation();
+
+  // Set the location data source so we use our GPS location as the originCamera.
+  m_arcGISArView->setOriginCamera(Camera());
+  m_arcGISArView->setTranslationFactor(1.0);
+
+  changeScene(true);
+}
+
+// Creates a scene based on the ImageryWithLabels base map.
+// Mode: Full-Scale AR
+void CppArExample::createImageryScene()
+{
+  // create a scene with imagery
+  m_scene = new Scene(BasemapType::ImageryWithLabels, this);
+  createSurfaceWithElevation();
+
+  // Set the location data source so we use our GPS location as the originCamera.
+  m_arcGISArView->setOriginCamera(Camera());
+  m_arcGISArView->setTranslationFactor(1.0);
+
+  changeScene(true);
+}
+
+// Creates a test scene
+// Mode: Full-Scale AR
+void CppArExample::createFullScaleTestScene()
+{
+  // create scene
```

Comment:
General comment: Capitalize the first word of each comment. Or don't. But choose one way to do it. I prefer capitals.